src/main/java/io/vertx/core/eventbus/impl/EventBusImpl.java

```diff
@@ -253,9 +257,13 @@ public EventBusMetrics<?> getMetrics() {
     return metrics;
   }
 
-  public MessageImpl createMessage(boolean send, String address, MultiMap headers, Object body, String codecName) {
+  public MessageImpl createMessage(boolean send, boolean localOnly, String address, MultiMap headers, Object body, String codecName) {
     Objects.requireNonNull(address, "no null address accepted");
-    MessageCodec codec = codecManager.lookupCodec(body, codecName, true);
+    boolean local = true;
+    if (vertx.isClustered()) {
```

Comment:
I think instead this should not be decided by this method and it should be passed by the caller, in other words the createMessage should just be dumb and apply what is said

Reply:
agreed, I will make the change